client/blocks/importer/wordpress/import-everything/migration-progress/use-restore-title.ts

```diff
@@ -0,0 +1,41 @@
+import { sprintf } from '@wordpress/i18n';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useEffect, useState } from 'react';
+
+export default function useBackupTitle( restorePercent?: number, restoreMessage?: string ) {
+	const translate = useTranslate();
+
+	const generateTitle = useCallback( () => {
+		const printArgs = {
+			percentage: Math.max( restorePercent || 0, 1 ),
+		};
+
+		if (
+			restoreMessage === 'Starting' ||
+			restoreMessage?.startsWith( 'Checking remote files' ) ||
+			restoreMessage?.startsWith( 'Parsing' )
```

Comment:
I don't like the way of checking the restoring message by checking if the message starts with a string.
Can we do something here? Can we provide a code for the current restore stage?

For example: `restoreStage: 'checking_remote_files' | 'parsing' | 'verifying' | 'uploading'`.

I think we should not rely on localized messages, which can be updated anytime and that potentially means regression on the FE side.

@zaerl, @ouikhuan, @vishnugopal, what do you folks think?

Reply:
These are strings that I collected by chance during the tests. Are output by Vaultpress API and are not localized. Unfortunately, the API outputs only three types of status: backup/restore/failed. The only other information we have from the API is this plain English message. Feel free to remove it.

Reply:
I think we still want to output as much information as we can, how about we handle these checks in the backend and map these as status codes ourselves? In this case, we can populate whatever we want in the FE with a localized string, and default to `Restoring up to x%` if we can't map the status?

Reply:
It is doable. The mapping can be performed in the same way I am doing it here in this prototype.

An ideal situation would be to change the VaultPress API to output more granular information. This will be the first concrete example of the improvement of external tools for migration.

Since we already have "Parsing," "Checking %s," etc. messages, it should be simple to identify different _phases_ and output this. More granular information and better it is when we have to do with an operation that can take hours to complete.